gcloud-java-dns/src/test/java/com/google/gcloud/dns/ZoneTest.java

```diff
@@ -0,0 +1,750 @@
+    assertNotNull(zone.dns());
+    assertEquals(dns, zone.dns());
+  }
+
+  @Test
+  public void testGetById() {
+    expect(dns.getZone(ZONE_ID)).andReturn(ZONE_INFO);
+    expect(dns.getZone(ZONE_ID, ZONE_FIELD_OPTIONS)).andReturn(ZONE_INFO); // for options
+    replay(dns);
+    Zone retrieved = Zone.get(dns, ZONE_ID);
+    assertSame(dns, retrieved.dns());
+    assertEquals(ZONE_INFO, retrieved.info());
+    BigInteger id = null;
+    try {
+      Zone.get(dns, id);
+      fail("Cannot get null zone.");
+    } catch (NullPointerException e) {
+      // expected
+    }
+    try {
+      Zone.get(null, id);
+      fail("Cannot get null zone.");
+    } catch (NullPointerException e) {
+      // expected
+    }
+    // test passing options
+    Zone.get(dns, ZONE_ID, ZONE_FIELD_OPTIONS);
+  }
+
+  @Test
+  public void testGetByName() {
+    expect(dns.getZone(ZONE_NAME)).andReturn(ZONE_INFO);
+    expect(dns.getZone(ZONE_ID, ZONE_FIELD_OPTIONS)).andReturn(ZONE_INFO); // for options
+    replay(dns);
+    Zone retrieved = Zone.get(dns, ZONE_NAME);
+    assertSame(dns, retrieved.dns());
+    assertEquals(ZONE_INFO, retrieved.info());
+    String name = null;
+    try {
+      Zone.get(dns, name);
+      fail("Cannot get null zone.");
+    } catch (NullPointerException e) {
+      // expected
+    }
+    try {
+      Zone.get(null, name);
+      fail("Cannot get null zone.");
+    } catch (NullPointerException e) {
+      // expected
+    }
+    // test passing options
+    Zone.get(dns, ZONE_ID, ZONE_FIELD_OPTIONS);
+  }
+
+  @Test
+  public void deleteByIdAndFound() {
+    expect(dns.delete(ZONE_ID)).andReturn(true);
+    replay(dns);
+    boolean result = zone.delete();
+    assertTrue(result);
+  }
+
+  @Test
+  public void deleteByIdAndNotFoundAndNameSetAndFound() {
+    expect(dns.delete(ZONE_ID)).andReturn(false);
+    expect(dns.delete(ZONE_NAME)).andReturn(true);
+    replay(dns);
+    boolean result = zone.delete();
+    assertTrue(result);
+  }
+
+  @Test
+  public void deleteByIdAndNotFoundAndNameSetAndNotFound() {
+    expect(dns.delete(ZONE_ID)).andReturn(false);
+    expect(dns.delete(ZONE_NAME)).andReturn(false);
+    replay(dns);
+    boolean result = zone.delete();
+    assertFalse(result);
+  }
+
+  @Test
+  public void deleteByIdAndNotFoundAndNameNotSet() {
+    expect(dns.delete(ZONE_ID)).andReturn(false);
+    replay(dns);
+    boolean result = zoneNoName.delete();
+    assertFalse(result);
+  }
+
+  @Test
+  public void deleteByNameAndFound() {
+    expect(dns.delete(ZONE_NAME)).andReturn(true);
+    replay(dns);
+    boolean result = zoneNoId.delete();
+    assertTrue(result);
+  }
+
+  @Test
+  public void deleteByNameAndNotFound() {
+    expect(dns.delete(ZONE_NAME)).andReturn(true);
+    replay(dns);
+    boolean result = zoneNoId.delete();
+    assertTrue(result);
+  }
+
+  @Test
+  public void listDnsRecordsByIdAndFound() {
+    Page<DnsRecord> pageMock = createStrictMock(Page.class);
```

Comment:
For here and bellow:
http://stackoverflow.com/questions/56954/easymock-how-do-i-create-a-mock-of-a-genericized-class-without-a-warning

Either of the following options is fine with me:
1) add `@SuppressWarnings("unchecked")` where needed
2) Use one the suggetions mentioned in SO
3) Add Page a simple "PageImpl" under https://github.com/GoogleCloudPlatform/gcloud-java/tree/master/gcloud-java-core/src/main/java/com/google/gcloud/testing, that gets one or more list of values.

I think Option (3) would be useful for you and others but I am fine if you choose any of the other options.